### ISSUE TITLE
Warns you that a not registered component is being used in the template

### ISSUE THREAD

**Issue by jesusgn90** [vuejs/eslint-plugin-vue#1112]
**Please describe what the rule should do:**

Warns you that a component is being used in the template and the component is not registered.

**What category should the rule belong to?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
   <some-component />
</template>

<script>
export default {
   name: 'Foo',
}
</script>
```

Being `SomeComponent` a component that was not registered.

**Additional context**

**Comment by sinaa** [comment]
This would be a really useful lint rule to have. 👍 

Current answer on StackOverflow ( https://stackoverflow.com/questions/57615609/can-i-fail-a-vue-js-build-on-unregistered-components ) suggests that there's no nice and simple way to get lint warning/failure when unregistered components are used.
